napari/_qt/exceptions.py

```diff
@@ -37,9 +40,25 @@ def handle(
         tb : TracebackType
             The traceback object associated with the error.
         """
-        # etype.__module__ contains the module raising the error
-        # Custom exception classes can have different behavior
-        # can add custom exception handlers here ...
         text = "".join(traceback.format_exception(etype, value, tb))
-        logging.error("Unhandled exception:\n%s", text)
         self.error.emit((etype, value, tb))
+        if session_is_interactive():
+            logging.error("Unhandled exception:\n%s", text)
+        else:
+            self._show_error_dialog(value)
+
+    def _show_error_dialog(self, exception: BaseException):
+        self.message = NapariNotification.from_exception(exception)
+        self.message.show()
+
+
+def session_is_interactive() -> bool:
```

Comment:
I feel like this should go in our main utils no? We might want to reuse this to fix #380 where we only call raise if session is interactive?

Reply:
sure